core/src/main/java/io/grpc/transport/AbstractStream.java

```diff
@@ -229,6 +278,74 @@ protected final void requestMessagesFromDeframer(int numMessages) {
     }
   }
 
+  /**
+   * Event handler to be called by the subclass when a number of bytes are being queued for sending
+   * to the remote endpoint. May call back the listener's {@link StreamListener#onReady()}
+   * handler if appropriate.
+   *
+   * @param numBytes the number of bytes being sent.
+   */
+  protected final void onSendingBytes(int numBytes) {
+    boolean shouldNotify = false;
+    synchronized (onReadyLock) {
+      shouldNotifyOnReady = true;
+      numSentBytesQueued += numBytes;
+      shouldNotify = shouldNotifyOnReady();
+    }
+    if (shouldNotify) {
+      listener().onReady();
+    }
+  }
+
+  /**
+   * Event handler to be called by the subclass when a number of bytes has been sent to the remote
+   * endpoint. May call back the listener's {@link StreamListener#onReady()} handler if appropriate.
+   *
+   * @param numBytes the number of bytes that were sent.
+   */
+  protected final void onSentBytes(int numBytes) {
+    boolean shouldNotify = false;
```

Comment:
Could we not initialize this variable here, since its value will always be overwritten?